src/API/Site/Controllers/Ads/RecommendationsController.php

```diff
@@ -0,0 +1,203 @@
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AccountService;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class RecommendationsController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads
+ */
+class RecommendationsController extends BaseController {
+
+	/**
+	 * Service used to access / update Ads account data.
+	 *
+	 * @var AccountService
+	 */
+	protected $account;
+
+	/**
+	 * RecommendationsController constructor.
+	 *
+	 * @param RESTServer     $server
+	 * @param AccountService $account
+	 */
+	public function __construct( RESTServer $server, AccountService $account ) {
+		parent::__construct( $server );
+		$this->account = $account;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'ads/recommendations',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_recommendations_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'type' => [
+				'type'        => 'string',
+				'description' => __( 'Filter recommendations by type', 'google-listings-and-ads' ),
+				'required'    => false,
+			],
+			'id'   => [
+				'type'        => 'integer',
+				'description' => __( 'Filter recommendations by unique id', 'google-listings-and-ads' ),
+				'required'    => false,
+			],
+		];
+	}
+
+	/**
+	 * Get the callback function for the list accounts request.
+	 *
+	 * @return callable
+	 */
+	protected function get_recommendations_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				// Checks if the ads account is connected; exits early if not connected to prevent further execution.
+				$account_status = $this->account->get_connected_account();
+				if ( isset( $account_status['status'] ) && 'connected' !== $account_status['status'] ) {
+					return new Response(
+						[ 'message' => __( 'No connected Ads account found.', 'google-listings-and-ads' ) ],
+						403
+					);
+				}
+
+				$type = $request->get_param( 'type' );
+
+				// Only support filtering by type IMPROVE_PERFORMANCE_MAX_AD_STRENGTH for now.
+				if ( $type && $type !== 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH' ) {
+					return new Response( [] );
+				}
```

Comment:
You can add an `enum` property to argument schema and the validator will automatically return an error when someone passes an unsupported value.

Ex:

```php
	public function get_collection_params(): array {
		return [
			'type' => [
				'type'        => 'string',
				'description' => __( 'Filter recommendations by type', 'google-listings-and-ads' ),
				// This could also use a callback to get the set of supported recommendation types from the `AdsRecommendations` service. 
				'enum'        => array( 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH' ),
				'required'    => false,
			],
			'id'   => [
				'type'        => 'integer',
				'description' => __( 'Filter recommendations by unique id', 'google-listings-and-ads' ),
				'required'    => false,
			],
		];
	}
```